File: SESION3/C++/Filtro.cpp
#include "Filtro.h"


Filtro::Filtro(){
        Ubicacion maracena("Polideportivo Maracena", 4.5, 0);
        Ubicacion granada("Campo de Futbol Granada", 20, 3);
        Ubicacion atarfe("Pista Tenis Atarfe",0,0);
        Ubicacion albolote("Pista de Tenis Albolote", 7, 1);
        Ubicacion illora("Pista de Futbol Illora", 5, 2);
        Ubicacion cogollos("Pista de Tenis de Cogollos", 6, 0);

        ubicaciones.push_back(maracena);
        ubicaciones.push_back(granada);    
        ubicaciones.push_back(atarfe);
        ubicaciones.push_back(albolote);
        ubicaciones.push_back(illora);
        ubicaciones.push_back(cogollos);

}

File: SESION3/dart/filtro.dart
import 'package:sesion_3_ds/ubicacion.dart';
import 'package:sesion_3_ds/alquiler.dart';

abstract class Filtro{


    List<Ubicacion> ubicaciones=[];

 
    Filtro(){
        Ubicacion maracena=new Ubicacion("Polideportivo Maracena", 4.5, 0);
        Ubicacion granada=new Ubicacion("Campo de Futbol Granada", 20, 3);
        Ubicacion atarfe=new Ubicacion("Pista Tenis Atarfe",0,0);
        Ubicacion albolote=new Ubicacion("Pista de Tenis Albolote", 7, 1);
        Ubicacion illora=new Ubicacion("Pista de Futbol Illora", 5, 2);
        Ubicacion cogollos=new Ubicacion("Pista de Tenis de Cogollos", 6, 0);

        ubicaciones.add(maracena);
        ubicaciones.add(granada);    
        ubicaciones.add(atarfe);
        ubicaciones.add(albolote);
        ubicaciones.add(illora);
        ubicaciones.add(cogollos);
    }
    
    void aplicarFiltro(Alquiler alq);
}